sections/monte_carlo/quantum_monte_carlo/quantum_monte_carlo.tex

```diff
@@ -97,71 +311,259 @@ \subsubsection{Construct the payoff function}\label{sec:MC_Payoff}
+              comparator.x(qubits_compare[0])
+          # uncompute ancillas state
+          for i in reversed(range(num_qubits_dist - 1)):
+              if i == 0:
+                  if bin_compare_value[i] == 1:
+                      comparator.cx(qubits_dist[i], qubits_compare[i+1])
+              else:
+                  if bin_compare_value[i] == 1:
+                      comparator.compose(OR(2), [qubits_dist[i], qubits_compare[i], qubits_compare[i+1]], inplace=True)
+                  else:
+                      comparator.ccx(qubits_dist[i], qubits_compare[i], qubits_compare[i+1])
+      else:
+          # num_state_qubits == 1 and value == 1:
+          comparator.cx(qubits_dist[0], qubits_compare[0])
+          # flip result bit if geq flag is false
+          if not bigger_then:
+              comparator.x(qubits_compare[0])
+  else:
+      if not bigger_then:  # otherwise the condition is never satisfied
+          comparator.x(qubits_compare[0])
+  comparator.name = name.format(value=compare_value, bit_value="".join(str(x) for x in bin_compare_value))
+  return comparator
+\end{lstlisting}
 
-Now that we have shown how to construct a piece-wise linear function in a quantum computer we change $\theta_P$ again to better obtain our desired result $\mathbb{E}$.
+\subsubsection{Implementation of piece-wise linear functions}\label{sec:ImplPieceFunc}
+
+Implementing the piecewise linear function is not directly straigth forward so a detailed introduction is given in the following. 
+As mentioned above we are dealing with linear function to encode the payoff
+\begin{equation}\label{eq:defLinearFunc}
+  f(x) = \alpha x + \beta,
+\end{equation}
+where $f(x)$ maps $x$ to the payoff by equation \ref{eq:callPayoff} and the domain of $x$ consist of of two linear function. One for $S_T < K$ and one for $S_T \geq K$. 
+Furthermore the $S_i$ do not occure as a real number but as an element of $\{ 0,...,2^n-1 \}$ and additionally on a quantum computer $f(x)$ has to be realised
+using rotations in the corresponding Hilbert space. 
+These challenges require us to use a affine mapping $\hat{f}(x)$ which is in contrast to
+\begin{equation}\label{eq:linFuncBound}
+  f: [ a,b ] \rightarrow [ c,d ]
+\end{equation}
+defined as
+\begin{equation}\label{eq:hatLinFunc}
+  \hat{f}: \{0,...,2^n-1 \} \rightarrow [ 0,1 ].
+\end{equation}
+Note that in this case $d=b-K$ thanks to the definition of the payoff. But how is the transformation from \ref{linFuncBound}
+to \ref{eq:hatLinFunc} actually done?\\
+For piecewise function there are of course breakpoints needed within $\{ 0,...,2^n-1 \}$
+\begin{equation}\label{eq:breakpoints}
+b_p(x) = \frac{x-a}{b-a} \times 2^n-1.
+\end{equation}
+In case of the payoff only one breakpoint is given for $ S_T \geq K $.
+The affine transformation from $f$ to $\hat{f}$ is then
+\begin{align}\label{eq:affinTransform}
+  \phi(x) &= a + \frac{b-a}{2^n-1} \cdot x \\
+  \hat{f}(x) &= \frac{f(\phi(x))-c}{d-c},
```

Comment:
Following the previous comment. This should then be  \hat{f}(x) &= 2\frac{f(\phi(x))-c}{d-c}-1

Later on when calculating the expectation, this is somehow what you already used as well. The outcome of P1 therefore does not need to change